array/array.go

```diff
@@ -195,7 +195,7 @@ func Slice(arr Interface, i, j int) Interface {
 	if arr, ok := arr.(sliceable); ok {
 		return arr.Slice(i, j)
 	}
-	if arr, ok := arr.(array.Interface); ok {
+	if arr, ok := arr.(arrow.Array); ok {
```

Comment:
So did arrow move around and rename types? I guess they had a complete breaking change from the older version,

Reply:
Interface aliases arrow.Array so that existing users don't get broken by the migration to arrow.Array. 
This alias will be removed in v8

https://pkg.go.dev/github.com/apache/arrow/go/v7@v7.0.0-20220201061922-bf0ee3f9c50e/arrow/array#Interface

Reply:
We have an interface defined at line 27 in this file that is named after the corresponding Arrow type. Can we change the name of that to `Array` as well? That is likely to cause some breaking changes in our Go API, but there are breaking changes anyway.

Reply:
I did change the name of the interface to `Array` to match the Arrow array Interface.

Regarding removing our `Interface` completely, the current [Arrow Array interface](https://github.com/apache/arrow/blob/master/go/arrow/array.go#L83) has 2 extra members `json.Marshaler` & `Data() ArrayData` so these needs to be implemented as well at our end of implementations. I have not analyzed the code to see how much work is needed for that. If that is something simple to do I can do that in this PR itself or we can open a separate issue for that.

Reply:
Sure will do. Thanks.